.github/workflows/tox-pytest.yml

```diff
@@ -0,0 +1,73 @@
+name: tox-pytest
+
+on: [ push, pull_request ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Compile Zenodo Datastore DOIs
+      run: grep -e '.*10\.\(5281\|5072\)/zenodo\..*' src/pudl/workspace/datastore.py | sed -e 's/",*$//g' | sed -e 's/^.*"//g' | sort > datastore-dois.txt
+
+    - uses: actions/cache@v2
+      id: cache-zenodo-datastore
+      with:
+        path: ~/pudl-work/data/
+        key: zenodo-datastore-${{ hashFiles('datastore-dois.txt') }}
+        restore-keys: |
+          zenodo-datastore-${{ hashFiles('datastore-dois.txt') }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get install libsnappy-dev
+        python -m pip install --upgrade pip
+        pip install tox
+
+    - name: Check for cached data
+      run: |
+        mkdir -p ~/pudl-work/data/
+        find ~/pudl-work/data/
+
+    - name: Run PyTest with Tox
+      env:
+        API_KEY_EIA: ${{ secrets.API_KEY_EIA }}
```

Comment:
how do these secrets work? did you "tell" actions the secret and then actions knows where to look for this?

Reply:
You store the secrets in the Github repository settings via the web.